### PR TITLE
ERT 3.0 with data flow mode

### PR DESCRIPTION
**Pull request by chienwei-lan**

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/ert_user.c [hunk 1]
```diff
@@ -445,7 +446,7 @@ static int ert_cfg_cmd(struct xocl_ert_user *ert_user, struct ert_user_command *
 	if (cmd_opcode(ecmd) != OP_CONFIG)
 		return -EINVAL;
 
-	if (major > 2) {
+	if (major > 3) {
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/ert_user.c:449]: isn't ert_user for ert v2 only?

**Review by maxzhen — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/ert_user.c (hunk 1 above)
Comment (chienwei-lan) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/ert_user.c:449]: yes, you are right

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_fdt.c [hunk 2]
```diff
@@ -347,23 +347,12 @@ static struct xocl_subdev_map subdev_map[] = {
...
-		.dev_name = XOCL_ERT_USER,
-		.res_array = (struct xocl_subdev_res[]) {
-			{.res_name = NODE_ERT_CQ_USER},
-			{NULL},
-		},
-		.required_ip = 1,
-		.flags = XOCL_SUBDEV_MAP_USERPF_ONLY,
-		.build_priv_data = ert_build_priv,
-		.devinfo_cb = NULL,
- 	},
```
Comment (mamin506) [inline src/runtime_src/core/pcie/driver/linux/xocl/xocl_fdt.c:361]: I think we still need XOCL_SUBDEV_ERT_USER on SSv2 shell. Am I right?

**Review by mamin506 — COMMENTED** [review]
Overall, this looks good. Just confusing when and how to create ert_user subdev for SSv2 platform.

**Review by maxzhen — APPROVED** [review]

**Review by mamin506 — APPROVED** [review]

**Pull request merged**